### PR TITLE
Add alpine-sh and alpine-bash container images

### PR DESCRIPTION
**Pull request by tiagolobocastro**
ci: update to accommodate new images
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    feat: add new tiny container images
    
    Adds alpine-sh and alpine-bash which are minimalist images
    These can be useful for small init containers
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    feat: support multiple images
    
    Support for multiple images (dockerfiles) during the make process
    Ensures make test does not leave exited containers
    Add make clean to delete buildx builder
    Removes the push scripts as those are no longer used
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
@Abhinandan-Purkait @niladrih  comments on the names please :D

**Review by sinhaashish — APPROVED** [review]

**Comment by niladrih** [comment]
IMO, the names shouldn't be tied to their purpose on the kubernetes' 'machine', i.e. as initContainers. It should be tied to the key libs, packages it contains. Here we don't have an entrypoint, or cmd, so we can't tie the name to that.
Maybe..
- for the bash one, it could be `alpine-bash` or `bash-musl`(this will make it clear that it uses busybox, musl and has bash)
- for the sh one, it's not necessarily just for sh, it's a default alpine, with busybox, etc. Maybe `alpine`. But with that name, it's confusing if we have just taken an alpine an retagged it with some other version.

Do we need the init-sh container? Should the bash one be enough for our use-cases at the moment? It adds another container to our helm chart and it saves very little space after we optimise it down.

**Comment by tiagolobocastro** [comment]
> IMO, the names shouldn't be tied to their purpose on the kubernetes' 'machine', i.e. as initContainers. It should be tied to the key libs, packages it contains. Here we don't have an entrypoint, or cmd, so we can't tie the name to that. Maybe..

hmm I'm not sure, calling it simply by the names of the packages feels somewhat incomplete.
That being said, I'm ok with alpine and alpine-bash as it's pretty clear what it is from the name, wdyt?
 
> Do we need the init-sh container? Should the bash one be enough for our use-cases at the moment? It adds another container to our helm chart and it saves very little space after we optimise it down.

The init-sh container is very simple to add since we adding the bash one, there's no cost of adding it so seemed a no brainer.

**Comment by niladrih** [comment]
I am okay with it. 

What about `helper-bash`? and that would change the old one to `helper-linux-utils`?

**Comment by tiagolobocastro** [comment]
> I am okay with it.
> 
> What about `helper-bash`? and that would change the old one to `helper-linux-utils`?

I guess so, though again we'd be adding purpose to the name :) 
I'd prefer `alpine-bash` and keep `linux-utils` as is for now. In the future we can come up with better name for the linux-utils since it has a lot more than just the linux-utils.

**Review by niladrih — COMMENTED** [review]
This'll require the image repositories to be created. I can take it up for docker.io, need a hand with quay. I'll try to see if my GitHub auth lets me work on ghcr.

File: Makefile.buildx.mk [hunk 1]
```diff
@@ -33,19 +33,33 @@ else
...
+	@docker buildx stop --builder container-builder
 	@echo
 
-.PHONY: buildx.push
-buildx.push:
-	BUILDX=true DIMAGE=${IMAGE_ORG}/linux-utils ./buildscripts/push
+.PHONY: buildx.clean
+buildx.clean:
+	docker buildx rm --builder container-builder || true
+	docker image prune -f
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply Makefile.buildx.mk:58]: No, prune only removes dangling layers, example.
Though actually it's not needed at this stage on the buildx, only on the non-buildx

<none>                       <none>    4f9bf1d07fda   14 hours ago   63.8MB

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -16,8 +16,15 @@ name: build
 on: ['push']
```
Comment (niladrih) [inline .github/workflows/build.yml:16]: This is not related to your change. I notice that the CI is trying to push to the container registry in the PR CI run. This is a huge problem. It should only push to the container registry when there is a push to the main branch, and main branch only.
```suggestion
on:
  push:
    branches:
      - 'develop'
```
```suggestion
on: ['push']
```

**Review by niladrih — APPROVED** [review]

File: dockerfiles/alpine-sh/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM alpine:3.20.1
```
Comment (Abhinandan-Purkait) [inline dockerfiles/alpine-sh/Dockerfile:1]: How about alpine:latest?

**Review by Abhinandan-Purkait — APPROVED** [review]

**Pull request merged**